brainfuck/executor.go

```diff
@@ -0,0 +1,54 @@
+package brainfuck
+
+func Interpret(program string) {
+	exec := compile(program)
+	exec.Execute()
+}
+
+func compile(input string) *program {
+	p := newParser()
+	result := newProgram(p.parse(input), newMemory())
```

Comment:
I would assign a result of this operation `p.parse(input)` to a separate variable to highlight that it returns a list of parsed commands